protocol/consistencychecks.go

```diff
@@ -224,12 +224,19 @@ func verifyAuthPath(uname string, key []byte,
 		key = ap.Leaf.Value
 	}
 
-	// verify auth path
-	if !ap.Verify([]byte(uname), key, str.TreeHash) {
-		return CheckBadAuthPath
+	if ap.ProofType() == m.ProofOfAbsence {
+		if !ap.VerifyIndex() {
+			return CheckBadLookupIndex
+		}
+	} else {
+		if !ap.VerifyBinding([]byte(uname), key) {
```

Comment:
Shouldn't you do the same as below and check `key != nil &&` and remove setting key above?